### ISSUE TITLE
DPatch Attack Error in `PyTorchObjectDetector`

### ISSUE THREAD

**Issue by f4str** [Trusted-AI/adversarial-robustness-toolbox#2248]
**Describe the bug**
When running the DPatch attack for the `PyTorchObjectDetector` and `PyTorchFasterRCNN`, an error can occur when using any PyTorch-specific preprocessors.

**To Reproduce**
Run the DPatch patch attack on the `PyTorchObjectDetector` while using a Faster R-CNN model. When using additional PyTorch-specific preprocessors, the error above will occur.

**Expected behavior**
The attacks should run without error. It can be easily fixed by adding the following line to the `_get_losses` method in the `PyTorchObjectDetector` class:

```python
if x_preprocessed.is_leaf:
    x_preprocessed.requires_grad = True
else:
    x_preprocessed.retain_grad()
```

This error occurs because the gradient retention property is lost when moving a tensor to a different device.

**Screenshots**

**System information (please complete the following information):**
 - OS
 - Python version
 - ART version or commit number
 - TensorFlow / Keras / PyTorch / MXNet version
